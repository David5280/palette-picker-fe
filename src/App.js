import React from 'react';
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <h1>Palette! Picker</h1>
      </header>
      <main>
        <section>
          <div>Palette 1</div>
          <div>Palette 2</div>
          <div>Palette 3</div>
          <div>Palette 4</div>
          <div>Palette 5</div>
          <button>Generate new Palette!</button>
          <form>
            <select>
              <option>Palette 1</option>
              <option>Palette 2</option>
              <option>Palette 3</option>
            </select>
            <input type='text' placeholder='Palette Name' />
            <button>Save Palette</button>
          </form>
        </section>
        <section>
          <form>
            <label for='new-project-name'>Create a new project:</label>
            <div>
              <input type='text' placeholder='New Project Name...' />
              <button>Save Project</button>
            </div>
          </form>
        </section>
        <section>
          <p>Project 1</p>
          <p>Project 2</p>
        </section>
      </main>
    </div>
  );
}

export default App;
