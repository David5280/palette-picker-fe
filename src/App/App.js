import React from 'react';
import Palette from '../Palette/Palette';
import ProjectContainer from '../ProjectContainer/ProjectContainer';
import { serverCall } from '../fetchCalls/fetchCalls';
import { connect } from 'react-redux';
import { retrieveProjects, loadComplete, hasErrored } from '../actions';

class App extends React.Component {
  constructor() {
    super()
    this.state = {}
  }

  componentDidMount = async () => {
    try {
      const projects = await serverCall('projects')
      await this.props.retrieveProjects(projects)
      await this.props.loadComplete()
    } catch (error) {
      this.props.loadComplete()
      this.props.hasErrored(error.message)
    }
  }

  setColors = (colors) => {
    this.setState({colors})
  }

  render() {
    return (
      <div className="App">
        <header>
          <h1>Palette! Picker</h1>
        </header>
        <main>
          <Palette />
          <ProjectContainer />
        </main>
        <footer>
          <p>Follow us on the interwebs, nerd.</p>
        </footer>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  retrieveProjects: projects => dispatch(retrieveProjects(projects)),
  loadComplete: () => dispatch(loadComplete()),
  hasErrored: error => dispatch(hasErrored(error))
});

export default connect(null, mapDispatchToProps)(App);
