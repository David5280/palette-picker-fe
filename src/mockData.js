export const mockProjects= [
  {
    "id": 1,
    "name": "Artisinal Cheese Shop",
    "created_at": "2019-08-25T21:04:48.793Z",
    "updated_at": "2019-08-25T21:04:48.793Z"
  },
  {
    "id": 3,
    "name": "Butcher Shop",
    "created_at": "2019-08-25T21:04:48.798Z",
    "updated_at": "2019-08-25T21:04:48.798Z"
  },
  {
    "id": 2,
    "name": "Barber",
    "created_at": "2019-08-25T21:04:48.798Z",
    "updated_at": "2019-08-25T21:04:48.798Z"
  }
]

export const singleMockProject = {
  "id": 1,
  "name": "Artisinal Cheese Shop",
  "created_at": "2019-08-25T21:04:48.793Z",
  "updated_at": "2019-08-25T21:04:48.793Z",
  "palette": [
    {
      "id": 1,
      "name": "Warm",
      "created_at": "2019-08-25T21:04:48.835Z",
      "updated_at": "2019-08-25T21:04:48.835Z",
      "color_1": "#000123",
      "color_2": "#900123",
      "color_3": "#J00123",
      "color_4": "#000123",
      "color_5": "#000123",
      "project_id": 1
    },
    {
      "id": 2,
      "name": "Cool",
      "created_at": "2019-08-25T21:04:48.836Z",
      "updated_at": "2019-08-25T21:04:48.836Z",
      "color_1": "#H00123",
      "color_2": "#A00123",
      "color_3": "#G00123",
      "color_4": "#670123",
      "color_5": "#000123",
      "project_id": 1
    }
  ]
}

export const mockPalettes = [
  {
    "id": 1,
    "name": "Warm",
    "color_1": "#000123",
    "color_2": "#900123",
    "color_3": "#J00123",
    "color_4": "#000123",
    "color_5": "#000123",
    "project_id": 1,
    "created_at": "2019-08-25T21:04:48.835Z",
    "updated_at": "2019-08-25T21:04:48.835Z"
  },
  {
    "id": 3,
    "name": "Oblique",
    "color_1": "#600123",
    "color_2": "#000123",
    "color_3": "#120123",
    "color_4": "#006123",
    "color_5": "#430123",
    "project_id": 3,
    "created_at": "2019-08-25T21:04:48.837Z",
    "updated_at": "2019-08-25T21:04:48.837Z"
  },
  {
    "id": 2,
    "name": "Cool",
    "color_1": "#H00123",
    "color_2": "#A00123",
    "color_3": "#G00123",
    "color_4": "#670123",
    "color_5": "#000123",
    "project_id": 1,
    "created_at": "2019-08-25T21:04:48.836Z",
    "updated_at": "2019-08-25T21:04:48.836Z"
  },
  {
    "id": 4,
    "name": "Winter",
    "color_1": "#000123",
    "color_2": "#000123",
    "color_3": "#120123",
    "color_4": "#005123",
    "color_5": "#430123",
    "project_id": 3,
    "created_at": "2019-08-25T21:04:48.840Z",
    "updated_at": "2019-08-25T21:04:48.840Z"
  },
  {
    "id": 5,
    "name": "Autumn",
    "color_1": "#010123",
    "color_2": "#043123",
    "color_3": "#430123",
    "color_4": "#000123",
    "color_5": "#300123",
    "project_id": 2,
    "created_at": "2019-08-25T21:04:48.841Z",
    "updated_at": "2019-08-25T21:04:48.841Z"
  },
  {
    "id": 6,
    "name": "Rustic",
    "color_1": "#500123",
    "color_2": "#000123",
    "color_3": "#120123",
    "color_4": "#800123",
    "color_5": "#430123",
    "project_id": 2,
    "created_at": "2019-08-25T21:04:48.842Z",
    "updated_at": "2019-08-25T21:04:48.842Z"
  },
  {
    "id": 7,
    "name": "Resonant",
    "color_1": "#600123",
    "color_2": "#060123",
    "color_3": "#120123",
    "color_4": "#003123",
    "color_5": "#431123",
    "project_id": 2,
    "created_at": "2019-08-25T21:04:48.843Z",
    "updated_at": "2019-08-25T21:04:48.843Z"
  }
]

export const singleMockPalette = {
  "id": 1,
  "name": "Warm",
  "color_1": "#000123",
  "color_2": "#900123",
  "color_3": "#J00123",
  "color_4": "#000123",
  "color_5": "#000123",
  "project_id": 1,
  "created_at": "2019-08-25T21:04:48.835Z",
  "updated_at": "2019-08-25T21:04:48.835Z"
}

export const mockColors = [
  '#fffg11',
  '#faf00g',
  '#fffg33',
  '#gsgf90',
  '#fhhggg',
]