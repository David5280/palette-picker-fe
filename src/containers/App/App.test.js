import React from 'react';
import {App, mapDispatchToProps, mapStateToProps} from './App.js';
import { shallow } from 'enzyme'
import { mockProjects } from '../../mockData.js';

describe('App Container', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <App 
        projects={mockProjects} 
        retrieveProjects={jest.fn()}
        loadComplete={jest.fn()}
        hasErrored={jest.fn()}
      />
    )

  })

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should do something', () => {

  })
})

