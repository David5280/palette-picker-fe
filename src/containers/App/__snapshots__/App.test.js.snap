// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Container renders without crashing 1`] = `ShallowWrapper {}`;
